includes/class-wc-stripe-webhook-handler.php

```
@@ -515,6 +515,201 @@ public function process_webhook_dispute_closed( $notification ) {
+		foreach ( $subscriptions as $subscription ) {
+			if ( 'inactive' === $mandate_status && $revocation_reason ) {
+				$subscription->update_status( OrderStatus::CANCELLED, __( 'Subscription cancelled due to mandate revocation.', 'woocommerce-gateway-stripe' ) );
+			} elseif ( 'inactive' === $mandate_status ) {
+				$subscription->update_status( OrderStatus::ON_HOLD, __( 'Subscription paused due to mandate becoming inactive.', 'woocommerce-gateway-stripe' ) );
+			} elseif ( 'active' === $mandate_status && $subscription->has_status( OrderStatus::ON_HOLD ) ) {
+				// 'active' is a WooCommerce Subscriptions status, not a core WooCommerce order status,
+				// so OrderStatus::ACTIVE does not exist. Using string literal instead.
+				$subscription->update_status( 'active', __( 'Subscription reactivated due to mandate becoming active.', 'woocommerce-gateway-stripe' ) );
+			}
+		}
+	}
```

PHPMD (2.15.0)

[warning] The method update_subscription_for_mandate() has a Cyclomatic Complexity of 10. The configured cyclomatic complexity threshold is 10. (undefined)

(CyclomaticComplexity)